src/TendoPay/Exceptions/TendoPay_Integration_Exception.php

```diff
@@ -23,5 +23,54 @@ class TendoPay_Integration_Exception extends \Exception {
 	 */
 	public function __construct( $message, \Throwable $previous = null ) {
 		parent::__construct( $message, 0, $previous );
+
+		try {
+			$this->report($message, $previous);
+		} catch(Exception $e) {}
+	}
+
+
+	private function report($message, $previous) {
+		ob_start();
+		print_r($previous). PHP_EOL;
+		$raw = ob_get_contents();
+		ob_end_clean();
+
+		$trace = (preg_match('/\[string:Exception:private\] => ([^#]+)/s', $raw, $match))
+			? $match[1] . PHP_EOL . $previous->getTraceAsString()
+			: $previous->getTraceAsString();
+
+		$info = [$message];
+		if (class_exists('WooCommerce')) {
```

Comment:
No need for this check. Tendopay requires woocommerce. If WC is disabled the plugin will not start. So if we get here, we already know WC is enabled.

Reply:
I'll remove it.